source/common/network/utility.cc

```diff
@@ -346,12 +346,14 @@ Address::InstanceConstSharedPtr Utility::getIpv6LoopbackAddress() {
                          new Address::Ipv6Instance("::1", 0, nullptr));
 }
 
-Address::InstanceConstSharedPtr Utility::getIpv4AnyAddress(uint32_t port) {
-  CONSTRUCT_ON_FIRST_USE(Address::InstanceConstSharedPtr, new Address::Ipv4Instance(port));
```

Comment:
I mess up with this. `CONSTRUCT_ON_FIRST_USE` will create static object for Ipv4Instance, so it is going to build the first time with the parameter passed in first time also.

Reply:
I found the same. 
My local patch is to always return a newly created shared ptr. That will utilize the map lookup rather than iterating the map

https://github.com/lambdai/envoy-dai/blob/debugbreak0204/source/common/network/utility.cc#L349-L355

Reply:
yea, that is.

I come up a fix https://github.com/envoyproxy/envoy/pull/19802, but free to submit one if you want to since you already spend time on it.

Thanks for helping on this again!